numpy/core/src/private/get_attr_string.h

```diff
@@ -1,37 +1,55 @@
 #ifndef __GET_ATTR_STRING_H
 #define __GET_ATTR_STRING_H
 
-static NPY_INLINE int
-_is_basic_python_type(PyObject * obj)
+static NPY_INLINE npy_bool
+_is_basic_python_type(PyTypeObject *tp)
 {
-    if (obj == Py_None ||
-            PyBool_Check(obj) ||
-            /* Basic number types */
+    PyTypeObject const * const known_types[] = {
```

Comment:
no those variables may come from a dynamic loading so they are not const as far as the compiler can tell.
adding static won't work for the same reason.
You could add a static init variable, but imo just using an if statement is nicer than modifiable global state.